### PR TITLE
feat(rbac): save and display role description in the frontend

### PR DESCRIPTION
**Pull request by AndrienkoAleksandr**
### What does this pull request do:

https://github.com/janus-idp/backstage-showcase/issues/898

P.S.: It would be nice to merge it for GA 1.1

### CODE REVIEW

File: plugins/rbac/src/hooks/useSelectedMembers.ts [hunk 1]
```diff
@@ -60,6 +64,7 @@ export const useSelectedMembers = (
   return {
     selectedMembers: data,
     members: membersArr,
+    roleDescription,
```
Comment (invincibleJai) [inline plugins/rbac/src/hooks/useSelectedMembers.ts:67]: will it make sense to return role metadata here and avoid the new hook `useRoleMetadata` or refactor role fetching to `useRoleMetadata` and consume it here

**Comment by invincibleJai** [comment]
Thanks @AndrienkoAleksandr for PR ❤️ 

/cc @debsmita1 
/cc @divyanshiGupta

File: plugins/rbac/src/hooks/useSelectedMembers.ts (hunk 1 above)
Comment (AndrienkoAleksandr) [reply plugins/rbac/src/hooks/useSelectedMembers.ts]: Done, I used second option.

**Comment by AndrienkoAleksandr** [comment]
@invincibleJai, I think I handled your feedback.

File: plugins/rbac/src/hooks/useSelectedMembers.ts [hunk 2]
```diff
@@ -10,23 +10,42 @@ import { SelectedMember } from '../components/CreateRole/types';
...
+
+  return {
+    loading,
+    role: Array.isArray(roles) ? roles[0] : undefined,
+    roleError: (roleError as Error) || {
+      name: (roles as Response)?.status,
+      message: `Error fetching the role. ${(roles as Response)?.statusText}`,
+    },
+  };
+};
```
Comment (divyanshiGupta) [inline plugins/rbac/src/hooks/useSelectedMembers.ts:35]: @AndrienkoAleksandr can you move this hook to a separate file?

**Comment by divyanshiGupta** [comment]
@AndrienkoAleksandr can you also squash your commits into one and align PR title with commit title.

File: plugins/rbac/src/hooks/useSelectedMembers.ts (hunk 2 above)
Comment (AndrienkoAleksandr) [reply plugins/rbac/src/hooks/useSelectedMembers.ts]: Done.

**Comment by AndrienkoAleksandr** [comment]
> Since now we have support for adding/editing description, should we have edit icon on About card similar to other cards which will directly take the user to 1st step of edit form?
 
 +1, but I'm not sure that I have time to do that in this pr.
  
 >  We have some other fields also in the about card so we can skip it also for the time being. But just wanted to put it out there.
 
I think I have backend issue https://issues.redhat.com/browse/RHIDP-1382?filter=-1 around this topic. And this pr's included update role?.metadata?.modifiedBy and role?.metadata?.lastModified. So when we will resolve backend issue, then backend will provided read only fields "modifiedBy" and "lastModified" in the metadata  and frontend will display them without other farther modifications.

**Comment by AndrienkoAleksandr** [comment]
> @AndrienkoAleksandr can you also squash your commits into one and align PR title with commit title?

Done.

**Comment by divyanshiGupta** [comment]
@AndrienkoAleksandr on not adding any description while creating role, user should see No description in About card but its not showing anything.


https://github.com/janus-idp/backstage-plugins/assets/20724543/4a7db736-95f2-4410-b1e3-08ddf2f455b2

File: plugins/rbac/src/hooks/useRole.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    error: roleError,
+  } = useAsync(async () => await rbacApi.getRole(roleEntityRef));
+
+  return {
+    loading,
+    role: Array.isArray(roles) ? roles[0] : undefined,
+    roleError: (roleError as Error) || {
+      name: (roles as Response)?.status,
+      message: `Error fetching the role. ${(roles as Response)?.statusText}`,
+    },
```
Comment (invincibleJai) [inline plugins/rbac/src/hooks/useRole.ts:29]: with this data and roleError will always be there i.e as below, ideally it should be there only if there is error.

when data is there, roleError looks like below

```
message: "Error fetching the role. undefined"
name:  undefined
```

Above change is not related to this PR so we can take this as an issue and get this improved

cc @divyanshiGupta

**Review by invincibleJai — APPROVED** [review]
/lgtm
